src/scales.js

```diff
@@ -116,25 +119,34 @@ function asOrdinalType(key) {
   return registry.get(key) === position ? "point" : "ordinal";
 }
 
-export function exposeScales(figure, scaleDescriptors) {
-  const scales = figure.scales = {};
+export function exposeScales(scaleDescriptors) {
```

Comment:
The name “exposeScales” no longer seems accurate. Can we separate the two functions, keeping exposeScales as it was and having a separate method for computing the legends?